Fix content sanitization logic

pedropalb opened the pull request:
The replacement of `<` and `>` was breaking some HTML text contents. For example, instead of `<br>` being rendered like an actual line break, it was being rendered as a literal "\<br\>" string.

This PR just removes the replacements. So, if there is a better way to do this, please let me know.

pedropalb committed:
Add option to disable sanitization of angle brackets

MihanEntalpo commented:
Hello! Thanks for your PR.
Replacement was made to not brake some other functionality, that I don't remember right now :)

You will face some problems without this replacement.
I've tried some different approaches, and all of them had some flaws.

I recommend to use built-in allure feature to create self-contained reports, that was made some time later after I've created this repo.

Option `--single-file` should do the tricks with allure report generator.

pedropalb commented:
Hi @MihanEntalpo,

> You will face some problems without this replacement.
I've tried some different approaches, and all of them had some flaws.

I see. With this change I was able to successfully generate the reports I needed. But, indeed, I did not tested it extensively, so I understand your concern. 

> I recommend to use built-in allure feature to create self-contained reports, that was made some time later after I've created this repo.
>
> Option --single-file should do the tricks with allure report generator.

I have used the `--single-file` option in `allure generate`.  But then I have a problem and that's where your project comes in handy: I have to patch the allure's HTML and CSS before converting to a single file HTML, so I can customize and simplify the reports to share with non-technical stakeholders.

As far as I'm aware of, the allure cli generates the reports and coverts to a single HTML at once. And I need it done in two separate steps so I can patch it in between.

Thank you!

MihanEntalpo commented:
Hello @pedropalb 

If your change allows to achieve what you need and doesn't give unpredicted results (at least in your case), you can make this behaviour selectable by command-line argument, so that users of the tool can select which mode is most useful for the specific task.

So, if you kindly add some command line switch, like `--disable-tags-escaping`, which would activate new behavior, and make a clarification on it in Readme, I'd be glad to accept your PR

pedropalb commented:
@MihanEntalpo sorry for the late response.

Just to let you know, I updated the code adding the proposed option.
But I could not test it yet since I don't have test assets here in my home PC (and I cannot submit PRs to GitHub from my work PC).

Do you have any set of test assets (allure reports) that I could download and use for validating the changes?
Otherwise, this week I will try to generate some and properly test the changes

MihanEntalpo commented:
Hello! Thanks for update! PR looks good, and if it could be run against some allure report without errors, I would merge it. Maybe I can access some reports tomorrow and check your version, if you can do it faster, feel free to let me know :)
Changes are looking good and all that can cause problems is some simple bug that I can't check for by looking into PR from the mobile phone. If code runs, it's okey to merge it.

(Or maybe I'll write some tests with OpenAi codex. This is the purpose on which LLM Agents are capable at least to some extent)

MihanEntalpo commented:
@pedropalb would you kindly pull last version of my repo, and run integration tests on your change?
I've added tests what run in docker (see README for clarification)

pedropalb commented:
Hi, @MihanEntalpo !

I updated the code and added one test in `test_dummy.py` using gemini cli. 
I quickly checked it and it seems to make sense.
Maybe tomorrow I will try to add more tests and check them more carefully (I'm in a rush now ).

pedropalb committed:
Fix artifacts removal issue caused by root ownership
